gossip/emitter/emitter.go

```diff
@@ -232,15 +233,12 @@ func (em *Emitter) getSortedTxs() *types.TransactionsByPriceAndNonce {
 }
 
 func (em *Emitter) EmitEvent() (*inter.EventPayload, error) {
-	if em.config.Validator.ID == 0 {
+	if em.config.Validator.ID == 0 || em.world.IsBusy() {
 		// short circuit if not a validator
 		return nil, nil
 	}
 	sortedTxs := em.getSortedTxs()
 
-	if em.world.IsBusy() {
```

Comment:
It may make sense to leave the check closer to `em.world.Lock()` to reduce the chance of locking the mutex when there's an active block processing. The reason why we bother with `IsBusy` check is because `em.world.Lock` will lock engine mutex and then wait until block processing is complete (if any). That will make engine mutex unavailable while we're waiting for completion of block processing (in a case if it's running)

Reply:
I get your point. Reverting my changes.